prov/rxm/src/rxm_cq.c

```diff
@@ -1429,13 +1455,17 @@ void rxm_ep_do_progress(struct util_ep *util_ep)
 					     deferred_conn_entry, conn_entry_tmp)
 			rxm_ep_progress_deferred_queue(rxm_ep, rxm_conn);
 	}
+
+	ofi_coll_ep_progress(&util_ep->ep_fid);
 }
 
 void rxm_ep_progress(struct util_ep *util_ep)
 {
 	ofi_ep_lock_acquire(util_ep);
 	rxm_ep_do_progress(util_ep);
 	ofi_ep_lock_release(util_ep);
+
+	ofi_coll_process_pending(&util_ep->ep_fid);
```

Comment:
We'll need to see how to handle progress, so that when collective support isn't requested (common case), we don't incur the overhead of checking empty lists and acquiring extra locks.